client/my-sites/woocommerce/dashboard/required-plugins-install-view.jsx

```diff
@@ -152,6 +158,48 @@ class RequiredPluginsInstallView extends Component {
 		}
 	};
 
+	createTimeoutTimer = ( durationSeconds ) => {
+		if ( this.timeoutTimer ) {
+			return;
+		}
+
+		this.timeoutTimer = window.setTimeout( this.timeoutElapsed, durationSeconds * 1000 );
+	};
+
+	timeoutElapsed = () => {
+		// These status means store setup is finished, not stalled.
+		if ( [ 'IDLE', 'DONESUCCESS', 'DONEFAILURE' ].includes( this.state.engineState ) ) {
+			return;
+		}
+
+		recordTrack( 'calypso_woocommerce_setup_timeout', {
+			engine_state: this.state.engineState,
+			to_activate: this.state.toActivate.join( ',' ),
+			to_install: this.state.toInstall.join( ',' ),
+			working_on: this.state.workingOn,
+			progress: this.state.progress,
+		} );
+
+		this.setState( {
+			engineState: 'DONETIMEOUT',
+		} );
+	};
+
+	destroyTimeoutTimer = () => {
+		if ( this.timeoutTimer ) {
+			window.clearTimeout( this.timeoutTimer );
+			this.timeoutTimer = false;
+		}
+	};
+
+	resetTimeoutTimer = ( durationInSeconds ) => {
```

Comment:
You could combine `createTimeoutTimer` and `resetTimeoutTimer` into a single function called `setTimeoutTimer`. This is definitely more of an opinion though -- feel free to disagree if you prefer the separate functions.

Reply:
Definitely, I think it's simpler and better